presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/AbstractSqlInvokedFunctionNamespaceManager.java

```diff
@@ -43,19 +46,24 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.collect.ImmutableList.toImmutableList;
 import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 public abstract class AbstractSqlInvokedFunctionNamespaceManager
         implements FunctionNamespaceManager<SqlInvokedFunction>
 {
     private final ConcurrentMap<FunctionNamespaceTransactionHandle, FunctionCollection> transactions = new ConcurrentHashMap<>();
 
+    private final Set<CatalogSchemaPrefix> servingNamespaces;
     private final LoadingCache<QualifiedFunctionName, Collection<SqlInvokedFunction>> functions;
     private final LoadingCache<SqlFunctionHandle, FunctionMetadata> metadataByHandle;
     private final LoadingCache<SqlFunctionHandle, ScalarFunctionImplementation> implementationByHandle;
 
     public AbstractSqlInvokedFunctionNamespaceManager(SqlInvokedFunctionNamespaceManagerConfig config)
     {
+        this.servingNamespaces = config.getServingNamespaces().stream()
```

Comment:
The serving function namespaces should come from the table rather than the config. The config provides initial values. Database has the source of truth. Also, the initial values is supposed to be a stop-gap for proper grant support.